dot/state/test_helpers.go

```diff
@@ -234,38 +232,3 @@ func AddBlocksToStateWithFixedBranches(t *testing.T, blockState *BlockState, dep
 		}
 	}
 }
-
-func generateBlockWithRandomTrie(t *testing.T, serv *Service,
```

Comment:
If its only used by tests it should probably be in a _test file right?

Reply:
> If its only used by tests it should probably be in a _test file right?

test_helpers.go is also meant only for test. I don't think, its contents gets used anywhere else. I do remember renaming some of them to say `helpers_test.go`, btw.

Reply:
Deepsource was pointing out that this test was not used anywhere else but this function is only used by `_integration_test.go` file that happens because it (deepsource) does not use `--tags=integration` to compile the code. To solve we need to move the function to the only integration file that uses it.

Reply:
I feel that in such situations when deepsource is not giving the right advice, we should ignore it. But I will leave that upto you and resolve this discussion.